src/providers/proxy/proxy_id.c

```diff
@@ -1542,6 +1688,11 @@ static int get_initgr_groups_process(TALLOC_CTX *memctx,
         }
     } while(status == NSS_STATUS_TRYAGAIN);
 
+    ret = remove_group_members(ctx, sysdb, dom, pwd, num_gids, gids);
```

Comment:
Do we really want to execute this function if `status` is other than `NSS_STATUS_SUCCESS` or `NSS_STATUS_NOTFOUND`?

Reply:
No, so I've moved the calls inside the `NSS_STATUS_SUCCESS` case.